### PR TITLE
Adding Test project for Azure Functions .NET 

### PR DESCRIPTION
**Pull request by issacnitin**

### CODE REVIEW

File: dotnet/aspnetcore/functionApp/Application/SampleFunctionApp.Test/TestFactory.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+using Microsoft.Extensions.Primitives;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace SampleFunctionApp.Test
+{
+    public class TestFactory
+    {
+        public static IEnumerable<object[]> Data()
```
[the remark this answers is not in the document]
Comment (issacnitin) [reply dotnet/aspnetcore/functionApp/Application/SampleFunctionApp.Test/TestFactory.cs]: I'm using it in FunctionsTest.cs file, to be passed to Azure Function's run method.

File: dotnet/aspnetcore/functionApp/Application/SampleFunctionApp.Test/ListLogger.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions.Internal;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace SampleFunctionApp.Test
+{
+    public class ListLogger : ILogger
```
[the remark this answers is not in the document]
Comment (issacnitin) [reply dotnet/aspnetcore/functionApp/Application/SampleFunctionApp.Test/ListLogger.cs]: This class is used to create a dummy ILogger implementation, for the Azure function's Run method

[the remark this answers is not in the document]
Comment (sachinma) [reply dotnet/aspnetcore/functionApp/Application/SampleFunctionApp.Test/ListLogger.cs]: But you are using nulllogger right now

File: dotnet/aspnetcore/functionApp/Application/SampleFunctionApp.Test/TestFactory.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (sachinma) [reply dotnet/aspnetcore/functionApp/Application/SampleFunctionApp.Test/TestFactory.cs]: I don't see it

File: dotnet/aspnetcore/functionApp/Application/SampleFunctionApp.Test/FunctionsTest.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+{
+    public class Function1Test
+    {
+        private readonly ILogger logger = TestFactory.CreateLogger();
+
+        [Fact]
+        public async void HttpTriggerWithParams()
+        {
+            var request = TestFactory.CreateHttpRequest("name", "Bill");
+            var response = (OkObjectResult)await Function1.Run(request, logger);
```
Comment (issacnitin) [inline dotnet/aspnetcore/functionApp/Application/SampleFunctionApp.Test/FunctionsTest.cs]: @sachinma I'm using the logger here. Passing null won't work

File: dotnet/aspnetcore/functionApp/Application/SampleFunctionApp.Test/TestFactory.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (issacnitin) [reply dotnet/aspnetcore/functionApp/Application/SampleFunctionApp.Test/TestFactory.cs]: I've tagged you in specific line..  Same as FakeTraceWriter for .net framework

[the remark this answers is not in the document]
Comment (issacnitin) [reply dotnet/aspnetcore/functionApp/Application/SampleFunctionApp.Test/TestFactory.cs]: I've updated, thanks!

**Pull request merged**